server/server.js

```diff
@@ -40,13 +40,17 @@ io.on("connection", (socket) => {
 
     console.log(`${userId} joined room ${roomId}`);
 
-    io.sockets.in(roomId).emit("userIsJoined", { users: users });
+    io.sockets.in(roomId).emit("userIsJoined", { users: data, success: true });
 
     socket
       .to(roomId)
       .emit("userJoinedRoom", { success: true, user: { socketId, ...data } });
   });
 
+  socket.on("getUserData", (data) => {
```

Comment:
there is no need to create this extra socket event for users in a room, we are getting it under usersIsJoined event as soon as user joins the room.